File: server/src/main/java/pl/bk/pizza/store/application/dto/order/DeliveryInfoDTO.java
package pl.bk.pizza.store.application.dto.order;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Getter
public class DeliveryInfoDTO
{
    private String name;
    private String surname;
    private AddressDTO addressDTO;
    private TelephoneDTO telephoneDTO;
}

File: client/pizza-store/src/app/order/deliveryInfoDTO.ts
import {AddressDTO} from "./addressDTO";
import {TelephoneDTO} from "./telephoneDTO";

export class DeliveryInfoDTO {

  name :String;
  surname :String;
  addressDTO :AddressDTO;
  telephoneDTO: TelephoneDTO;

  constructor(data:any) {
    this.name = data.name;
    this.surname = data.surname;
    this.addressDTO = new AddressDTO(data.address);
    this.telephoneDTO = new TelephoneDTO(data.telephone);
  }
}
